test/OpenTelemetry.AotCompatibility.Tests/AotCompatibilityTests.cs

```diff
@@ -0,0 +1,78 @@
+// <copyright file="AotCompatibilityTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace OpenTelemetry.AotCompatibility.Tests
+{
+    public class AotCompatibilityTests
+    {
+        private readonly ITestOutputHelper testOutputHelper;
+
+        public AotCompatibilityTests(ITestOutputHelper testOutputHelper)
+        {
+            this.testOutputHelper = testOutputHelper;
+        }
+
+        /// <summary>
+        /// This test ensures that the intended APIs of the OpenTelemetry.AotCompatibility.TestApp libraries are
+        /// trimming and NativeAOT compatible.
+        ///
+        /// This test follows the instructions in https://learn.microsoft.com/dotnet/core/deploying/trimming/prepare-libraries-for-trimming#show-all-warnings-with-sample-application
+        ///
+        /// If this test fails, it is due to adding trimming and/or AOT incompatible changes
+        /// to code that is supposed to be compatible.
+        ///
+        /// To diagnose the problem, inspect the test output which will contain the trimming and AOT errors. For example:
+        ///
+        /// error IL2091: 'T' generic argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicConstructors'.
+        /// </summary>
+        [Fact]
```

Comment:
Curious - would it make sense to have the AOT test build as a separate CI job instead of using unit test here? (easier to turn on/off, can run in parallel with existing unit test pipelines, etc.)

Reply:
I'm thinking while we are still developing, i.e. the warnings are > 0, we keep this as a unit test so that it's easier to capture the expected number of warnings and run the test locally.
Once we fixed/suppressed all the warnings, we can bring it to be a CI to get the above mentioning benefits.

Reply:
> that it's easier to capture the expected number of warnings and run the test locally.

You should be able to run the CI check locally as well.

Reply:
Sounds good to me 👍